javascript/javascript1/week2/main.js

```diff
@@ -144,3 +144,43 @@ function getNumberOfStudents() {
   console.log(studentStrength);
 }
 getNumberOfStudents();
+
+//Candy helper
+
+let boughtCandyPrices = [];
+let amountToSpend = Math.random() * 100;
+console.log("Amount to spend: " + amountToSpend);
+
+function addcandy(candyType, weight) {
+  switch (candyType) {
+    case "Sweet":
+      boughtCandyPrices.push(weight * 0.5);
+      break;
+    case "Chocolate":
+      boughtCandyPrices.push(weight * 0.7);
+      break;
+    case "Tofee":
+      boughtCandyPrices.push(weight * 1.1);
+      break;
+    case "Chewing-gum":
+      boughtCandyPrices.push(weight * 0.03);
+      break;
+  }
+}
+function canBuyMoreCandy() {
+  let totalBoughtCandyPrices = 0;
+  for (let i = 0; i < boughtCandyPrices.length; i++) {
+    totalBoughtCandyPrices = totalBoughtCandyPrices + boughtCandyPrices[i];
+  }
+  console.log("Total amount: " + totalBoughtCandyPrices);
+  if (totalBoughtCandyPrices < amountToSpend) {
+    console.log("You can buy more, so please do!");
+  } else {
+    console.log("Enough candy for you!");
+  }
+}
+
+addcandy("Sweet", 20);
+canBuyMoreCandy();
+addcandy("Chocolate", 10);
+canBuyMoreCandy();
```

Comment:
Good looking implementation ✨

Consider if your function canBuyMoreCandy should return values, in this implementation, you are logging. 💡
Lastly, reflect on whether your variables boughtCandyPrices and amountToSpend could be initialized as const. Check out this article about JavaScript and array const, if your are interested => https://www.w3schools.com/js/js_array_const.asp

Reply:
Thanks for pointing it out. I did it throughout the whole homework. I have changed it and pushed the modified version.